app/src/main/java/ca/bc/gov/bchealth/ui/custom/DatePickerExtension.kt

```diff
@@ -0,0 +1,70 @@
+package ca.bc.gov.bchealth.ui.custom
+
+import androidx.core.widget.doOnTextChanged
+import androidx.fragment.app.Fragment
+import ca.bc.gov.bchealth.R
+import ca.bc.gov.bchealth.utils.adjustOffset
+import com.google.android.material.datepicker.MaterialDatePicker
+import com.google.android.material.textfield.TextInputLayout
+import java.text.SimpleDateFormat
+import java.util.Locale
+
+/*
+* @author amit_metri on 13,January,2022
+*/
+
+val simpleDateFormat = SimpleDateFormat("yyyy-MM-dd", Locale.CANADA)
+
+fun Fragment.setUpDatePickerUi(tipDatePicker: TextInputLayout, tag: String) {
+
+    val dateOfBirthPicker =
+        MaterialDatePicker.Builder.datePicker()
+            .setTitleText(getString(R.string.select_date))
+            .setSelection(MaterialDatePicker.todayInUtcMilliseconds())
+            .build()
+    tipDatePicker.editText?.setOnClickListener {
+        dateOfBirthPicker.show(parentFragmentManager, tag)
+    }
+    tipDatePicker.setEndIconOnClickListener {
+        dateOfBirthPicker.show(parentFragmentManager, tag)
+    }
+    dateOfBirthPicker.addOnPositiveButtonClickListener {
+        tipDatePicker.editText?.setText(simpleDateFormat.format(it.adjustOffset()))
+    }
+}
+
+fun Fragment.validateDatePickerData(
+    tipDatePicker: TextInputLayout,
+    errorMessage: String
+): Boolean {
+    if (tipDatePicker.editText?.text.isNullOrEmpty()) {
+        tipDatePicker.isErrorEnabled = true
+        tipDatePicker.error = errorMessage
+        tipDatePicker.editText?.doOnTextChanged { text, _, _, _ ->
+            if (text != null && text.isNotEmpty()) {
+                tipDatePicker.isErrorEnabled = false
+                tipDatePicker.error = null
+            }
+        }
+        return false
+    }
+
+    if (!tipDatePicker.editText?.text.toString()
```

Comment:
These regexes are a bit dense. I would recommend pulling these conditions out and naming them separately. You could also save some repetition this way too (ignore formatting):
```
val isMissing by lazy { tipDatePicker.editText?.text.isNullOrEmpty() }
val validDate by lazy { tipDatePicker.editText?.text.toString().matches(Regex("^\\d{4}-\\d{2}-\\d{2}$")) }
val validDateTime by lazy { tipDatePicker.editText?.text.toString()
            .matches(Regex("^(\\d{4})-(0?[1-9]|1[012])-(0?[1-9]|[12][0-9]|3[01])$")) }
if (isMissing || !validDate || !validDateTime) {
   ...
    tipDatePicker.error = if (isMissing) errorMessage else getString(R.string.enter_valid_date_format)
}
```

Reply:
Will take care in gradual refactoring